src/NexusMods.App.UI/Controls/StandardButton.axaml.cs

```diff
@@ -0,0 +1,285 @@
+
+    /// <summary>
+    /// Defines the ShowLabel attached property of the <see cref="StandardButton"/>.
+    /// </summary>
+    public static readonly AttachedProperty<bool> ShowLabelProperty = AvaloniaProperty.RegisterAttached<StandardButton, TemplatedControl, bool>("ShowLabel", defaultValue: true);
+
+    /// <summary>
+    /// Gets or sets the text of the <see cref="StandardButton"/>.
+    /// </summary>
+    public string? Text
+    {
+        get => GetValue(TextProperty);
+        set => SetValue(TextProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the icon display option of the <see cref="StandardButton"/>.
+    /// </summary>
+    public ShowIconOptions? ShowIcon
+    {
+        get => GetValue(ShowIconProperty);
+        set => SetValue(ShowIconProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the left icon of the <see cref="StandardButton"/>.
+    /// </summary>
+    public IconValue? LeftIcon
+    {
+        get => GetValue(LeftIconProperty);
+        set => SetValue(LeftIconProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the right icon of the <see cref="StandardButton"/>.
+    /// </summary>
+    public IconValue? RightIcon
+    {
+        get => GetValue(RightIconProperty);
+        set => SetValue(RightIconProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the label is shown on the <see cref="StandardButton"/>.
+    /// </summary>
+    public bool ShowLabel
+    {
+        get => GetValue(ShowLabelProperty);
+        set => SetValue(ShowLabelProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the type of the <see cref="StandardButton"/>.
+    /// </summary>
+    public Types Type
+    {
+        get => GetValue(TypeProperty);
+        set => SetValue(TypeProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the size of the <see cref="StandardButton"/>.
+    /// </summary>
+    public Sizes Size
+    {
+        get => GetValue(SizeProperty);
+        set => SetValue(SizeProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the fill option of the <see cref="StandardButton"/>.
+    /// </summary>
+    public Fills Fill
+    {
+        get => GetValue(FillProperty);
+        set => SetValue(FillProperty, value);
+    }
+
+    /// <inheritdoc/>
+    protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+    {
+        base.OnApplyTemplate(e);
+
+        _leftIcon = e.NameScope.Find<UnifiedIcon>("PART_LeftIcon");
+        _rightIcon = e.NameScope.Find<UnifiedIcon>("PART_RightIcon");
+        _label = e.NameScope.Find<TextBlock>("PART_Label");
+        _content = e.NameScope.Find<ContentPresenter>("PART_ContentPresenter");
+        _border = e.NameScope.Find<Border>("PART_Border");
+
+        if (_leftIcon == null || _rightIcon == null || _label == null || _content == null || _border == null) return;
```

Comment:
At first I didn't understand why this was necessary, but I get it I think, because the Template is being set by the Theme or the style, so we can't just use the names directly like we would in `UserControls` code behinds

Reply:
This was mimicking how it was done in the Avalonia source and yes, as you've said, it could change by another Theme and so I think Avalonia needs to rebuild the references.